Lib/pdb.py

```diff
@@ -1335,29 +1351,85 @@ def do_list(self, arg):
         breaklist = self.get_file_breaks(filename)
         try:
             lines = linecache.getlines(filename, self.curframe.f_globals)
+            instructions = dis.get_instructions(self.curframe.f_code) \
```

Comment:
```suggestion
            instructions = dis.get_instructions(self.curframe.f_code, adaptive=True) \
```
I am not sure, but probably we need to ask for _specialized_ bytecode that reflects the bytecode that has been executed at the moment.